plugins/speculation-rules/settings.php

```diff
@@ -262,14 +266,38 @@ function plsr_render_settings_field( array $args ): void {
 			return; // @codeCoverageIgnore
 	}
 
-	$value = $option[ $args['field'] ];
+	$value                 = $option[ $args['field'] ];
+	$has_object_cache      = wp_using_ext_object_cache();
+	$authenticated_options = array( 'logged_out_and_admins', 'any' );
+	$show_warning          = 'authentication' === $args['field'] && ! $has_object_cache && in_array( $value, $authenticated_options, true );
 	?>
+	
+	<?php if ( 'authentication' === $args['field'] && ! $has_object_cache ) : ?>
+		<div id="plsr-auth-warning" class="notice notice-warning inline" <?php echo $show_warning ? '' : 'hidden'; ?>>
+			<p>
+				<strong><?php esc_html_e( 'Warning:', 'speculation-rules' ); ?></strong>
+				<?php
+				printf(
+					/* translators: %s: URL to persistent object cache documentation */
+					esc_html__( 'Enabling speculative loading for authenticated users without a persistent object cache may significantly increase server load. Consider setting up a %s before enabling this feature for logged-in users.', 'speculation-rules' ),
```

Comment:
This may be easier for translators, since "persistent object cache" appears inline with the sentence it is related to. For example, in some languages, "persistent object cache" would be translated differently if it is a subject or an object.

```php
__( 'Enabling speculative loading for authenticated users without a persistent object cache may significantly increase server load. Consider setting up a <a href="%s" target="_blank">persistent object cache</a> before enabling this feature for logged-in users.', 'speculation-rules' ),
```

Reply:
Examples: https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop%20%2F__%5C(.%2B%3F%3Ca%2F&type=code